### PR TITLE
Type definitions for catalog (2.x)

### PR DESCRIPTION
**Pull request by wereHamster**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not provide its own types, and you can not add them.
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

dts-gen was a really bad starting point. Instead, it's based on a type definition file we've used internally for a while.

### CODE REVIEW

File: types/catalog/catalog-tests.ts [hunk 1]
```diff
@@ -0,0 +1 @@
+import { render } from "catalog";
```
Comment (plantain-00) [inline types/catalog/catalog-tests.ts]: I think some test should be added in this file, you may refer to: https://docs.catalog.style/configuration

**Review by plantain-00 — CHANGES_REQUESTED** [review]

**Review by plantain-00 — APPROVED** [review]

**Comment by wereHamster** [comment]
@plantain-00 what are the next steps to get this PR merged?

**Comment by plantain-00** [comment]
Others will merge this, I just do code review

File: types/catalog/tslint.json [hunk 2]
```diff
@@ -0,0 +1,5 @@
+{
+	"extends": "dtslint/dt.json",
+	"rules": {
```
Comment (sheetalkamat) [inline types/catalog/tslint.json]: Please remove the rules from this tslint. You should just be extending dtslint/dt.json

**Review by sheetalkamat — CHANGES_REQUESTED** [review]
Please update the tslint file

**Review by sheetalkamat — APPROVED** [review]

**Pull request merged**